### PR TITLE
close #16569: deprecated `reversed(a, start, last)` overload, use toOpenArray instead

### LINKED ISSUE #16569: Change signature of `reversed`
The [`reversed`](https://nim-lang.github.io/Nim/algorithm.html#reversed%2CopenArray%5BT%5D%2CNatural%2Cint) proc (from the `algorithm` module) has the following signature:
```nim
proc reversed[T](a: openArray[T]; first: Natural; last: int): seq[T]
```
Notably, the type of `last` is `int` and not `Natural`, like in the [`reverse`](https://nim-lang.github.io/Nim/algorithm.html#reverse%2CopenArray%5BT%5D%2CNatural%2CNatural) proc.

Would it be ok to change the signature for consistency? Note that changing change the type to `Natural` requires changing the implementation of the other [`reversed`](https://nim-lang.github.io/Nim/algorithm.html#reversed%2CopenArray%5BT%5D) proc to:
```nim

proc reversed*[T](a: openArray[T]): seq[T] =
  reversed(a, 0, max(0, a.high))
```

### PR DESCRIPTION
**Pull request by timotheecour**
close #16569

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -76,6 +76,9 @@
 - `strformat` is now part of `include std/prelude`.
 
 
+- in `std/algorithm`, deprecated `proc reversed*[T](a: openArray[T], first: Natural, last: int): seq[T]`
```
Comment (ringabout) [inline changelog.md]:
```suggestion
- Deprecated `proc reversed*[T](a: openArray[T], first: Natural, last: int): seq[T]` in `std/algorithm`.
```

Comment (ringabout) [reply changelog.md]: or something else

**Review by ringabout — APPROVED** [review]

File: changelog.md (hunk 1 above)
Comment (timotheecour) [reply changelog.md]: ok, done

File: changelog.md [hunk 2]
```diff
@@ -80,6 +80,10 @@
   other defines.
 
 
+- in `std/algorithm`, deprecated `proc reversed*[T](a: openArray[T], first: Natural, last: int): seq[T]`
```
Comment (ringabout) [inline changelog.md]: Well, again :P

Comment (timotheecour) [reply changelog.md]: because `merge=union`, fixed :)

**Comment by Varriount** [comment]
Does toOpenArray do bounds checking?

**Comment by timotheecour** [comment]
> Does toOpenArray do bounds checking?

yes, PR welcome to document it via runnableExamples + doAssertRaises (even though it's not a surprising behavior)

**Pull request merged**